### ISSUE TITLE
Question about specifying range for 'provides direct input for'

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/GO_Shapes#19]
The MF to MF relation:

provides_direct_input_for: `@<MolecularFunction>` {0,1};

should only apply to a subset of MF terms, namely MF GO:0003824 'catalytic activity' to MF GO:0003824 'catalytic activity'.

How can we denote this in the specs?

**Comment by balhoff** [comment]
This is interesting—it would be straightforward if we were working directly with ontology subsumption rather than biolink categories. @cmungall I'm guessing there wouldn't be a category for this?

**Comment by goodb** [comment]
Yeah..  if we start building shapes that go deeper into the ontology, the advantages of the categories start to disappear and the argument goes back towards working over the entailed graph directly. I think this depends a lot on our predictions for how the shapes will be used.  

Another thing to consider (again) is the relation between the shapes and the restrictions in the ontology.  Like many cases, the above could be captured by changing the range of the OWL property definitions.  The Arachne consistency check could then detect and report the violation.  

@vanaukenk I think you denote this in the spec, for now, by creating a new shape called <CatalyticActivity> as a sub shape of MolecularFunction and moving the appropriate causal relation restrictions (e.g. provides direct input for) to that shape.  Then it falls on software team to figure out how to get that to work (by adding new tags or otherwise).

**Comment by pgaudet** [comment]
We will probably need use this type of pattern often (for example also for the happens_during relation). 

Do we need to make a decision WRT the best way to go about doing this now ? Or will we be able to 'easily' switch later if needed?

**Comment by goodb** [comment]
@pgaudet everything is possible and nothing is easy.. :)
I don't think it will be a showstopper to rework the structure of the validation framework if we need to.

**Comment by cmungall** [comment]
Remember, for any rule that is about biology rather than what structures we
can assert, we can use OWL axioms.

I would also be inclined to punt on this in the shex until after the MF
refactor, since we may be using this rule to patch the fact that the top
level of MF is still a bit of an odd grab bag

**Comment by pgaudet** [comment]
How will the MF help ?
